packages/next/client/components/reducer.ts

```diff
@@ -671,6 +720,20 @@ function clientReducer(
       const href = createHrefFromUrl(url)
       const pendingPush = navigateType === 'push'
 
+      // Do a full page navigation when the root layout changes
+      if (
+        mutable.patchedTree &&
+        !findRootLayoutInFlightRouterState(
+          mutable.patchedTree,
+          state.rootLayoutSegments
+        )
+      ) {
+        window.location.href = mutable.canonicalUrlOverride
```

Comment:
Do the full page nav here or set mpaNav to true?

Reply:
Set `mpaNavigation` to true because these can be discarded if you click on another link quickly.